dev/service-registry/devmode-scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/registry/ServiceRegistration.scala

```diff
@@ -6,19 +6,25 @@ package com.lightbend.lagom.internal.scaladsl.registry
 import java.net.URI
 
 import com.lightbend.lagom.scaladsl.api.ServiceInfo
+import com.typesafe.config.Config
 import play.api.inject.ApplicationLifecycle
 import play.api.{ Configuration, Logger }
 
 import scala.concurrent.{ ExecutionContext, Future }
 import scala.util.{ Failure, Success }
 
-class ServiceRegistration(serviceInfo: ServiceInfo, lifecycle: ApplicationLifecycle, config: Configuration,
+class ServiceRegistration(serviceInfo: ServiceInfo, lifecycle: ApplicationLifecycle, config: Config,
                           registry: ServiceRegistry)(implicit ec: ExecutionContext) {
 
+  @deprecated(message = "prefer constructor using typesafe Config instead", since = "1.4.0")
```

Comment:
This class is an internal class that is only used in the development environment, not a published API. So, it isn't necessary to keep the deprecated constructor: just replace all of the uses of the old constructor and remove it.

In general, it's usually safe to make incompatible changes to anything in an `internal` package.